src/tasks/MonoTargetsTasks/EmitBundleTask/EmitBundleBase.cs

```diff
@@ -0,0 +1,461 @@
+
+    public override bool Execute()
+    {
+        List<ITaskItem> bundledResources = new(FilesToBundle.Length);
+        foreach (ITaskItem bundledResource in FilesToBundle)
+        {
+            var resourcePath = bundledResource.ItemSpec;
+
+            bundledResource.SetMetadata("ResourceType", "DataResource");
+            try
+            {
+                using FileStream resourceContents = File.OpenRead(resourcePath);
+                using PEReader resourcePEReader = new(resourceContents);
+                if (resourcePEReader.HasMetadata)
+                {
+                    string? managedAssemblyCulture = null;
+
+                    var resourceMetadataReader = PEReaderExtensions.GetMetadataReader(resourcePEReader);
+                    if (resourceMetadataReader.IsAssembly)
+                    {
+                        bundledResource.SetMetadata("ResourceType", "AssemblyResource");
+                        managedAssemblyCulture = resourceMetadataReader.GetString(resourceMetadataReader.GetAssemblyDefinition().Culture);
+                    }
+
+                    bool isSatelliteAssembly = !string.IsNullOrEmpty(managedAssemblyCulture) && !managedAssemblyCulture!.Equals("neutral", StringComparison.OrdinalIgnoreCase);
+                    if (resourcePath.EndsWith(".resources.dll", StringComparison.InvariantCultureIgnoreCase) || isSatelliteAssembly)
+                    {
+                        bundledResource.SetMetadata("ResourceType", "SatelliteAssemblyResource");
+                        if (isSatelliteAssembly) {
+                            bundledResource.SetMetadata("Culture", managedAssemblyCulture);
+                        }
+                    }
+                }
+            }
+            catch (BadImageFormatException e)
+            {
+                if (resourcePath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+                    Log.LogMessage(MessageImportance.High, $"Resource '{resourcePath}' was interpreted with ResourceType 'DataResource' but has a '.dll' extension. Error: {e}");
+            }
+
+            var registeredName = bundledResource.GetMetadata("RegisteredName");
+            if (string.IsNullOrEmpty(registeredName))
+            {
+                string culture = bundledResource.GetMetadata("Culture");
+                if (!string.IsNullOrEmpty(culture))
+                {
+                    registeredName = culture + "/" + Path.GetFileName(resourcePath);
+                }
+                else
+                {
+                    registeredName = Path.GetFileName(resourcePath);
+                }
+                bundledResource.SetMetadata("RegisteredName", registeredName);
+            }
+
+            string resourceDataSymbol = $"bundled_resource_{ToSafeSymbolName(TruncateEncodedHash(Utils.ComputeHashEx(resourcePath, Utils.HashAlgorithmType.SHA256, Utils.HashEncodingType.Base64Safe), MaxEncodedHashLength))}";
```

Comment:
The output filename should be more descriptive though, like:
```csharp
            string destinationFileName = $"{Path.GetFileName(resourcePath)}_{ToSafeSymbolName(TruncateEncodedHash(Utils.ComputeHashEx(resourcePath, Utils.HashAlgorithmType.SHA256, Utils.HashEncodingType.Base64Safe), MaxEncodedHashLength))}_bundle{GetOutputFileExtension()}";
```

Reply:
The problem here is that we share identical bundled assets so we just generate one bundled item instead of duplicates, that means multiple different resources will map the same destination file, so that is why we can't use the file name as part of the file name, only the hash. Inside the generated file (if its a source file) there is a list of all resources that maps into the same destination file.

Reply:
`_resourcesForDataSymbolDictionary` tracks which resource registered names have the same file contents and therefore have the same symbol names. I'll add logging around `_resourcesForDataSymbolDictionary` logic to note when multiple resources have been mapped to the same destination file.